coordinator/points_writer.go

```diff
@@ -375,7 +392,11 @@ func (w *PointsWriter) WritePointsPrivileged(writeCtx tsdb.WriteContext, databas
 	atomic.AddInt64(&w.stats.SubWriteOK, 1)
 
 	if err == nil && len(shardMappings.Dropped) > 0 {
-		err = tsdb.PartialWriteError{Reason: "points beyond retention policy", Dropped: len(shardMappings.Dropped)}
+		err = tsdb.PartialWriteError{Reason: "points beyond retention policy or outside permissible write window",
```

Comment:
Looking below there the Points get looped through. Can there be multiple errors, should we be collecting all possible errors and returning them using an errors slice? Or is it expected to return on single error.

Reply:
We return the first error, because error wrapping with `errors.Join` is Go-specific, and we want a simple, single error that is accessible in all clients (`curl`, `python`, etc.